src/Ledger/Deleg.lagda

```diff
@@ -86,6 +91,7 @@ private variable
   pp : PParams
   vs : List GovVote
   poolParams : PoolParams
+  wdrls  : RwdAddr ⇀ Coin
```

Comment:
This looks fine.  My only concern is probably not a real concern... but I'll mention in case it's worth considering at some point.

Are long lists of private variables (hidden in the pdf doc) good practice, or at least innocuous?  Probably there's no risk that private variables could confuse the pdf reader, but I seem to recall reading in some Agda style recommendation document that, generally speaking, private variables are best limited to simple or very general type or level variables; e.g.,

```agda
private variable
  ℓ : Level
  a : A
  b : B
```
...and inhabitants of more special types should be explicit (as function arguments or module parameters).  On the other hand, I suppose if we are trying to make the pdf as concise and readable as possible, then I suppose using private variables very liberally like this makes more sense for us than for the typical Agda project.

Reply:
So the issue is that we're doing pretend pattern-matching in the STS constructors. `e ⊢  s ⇀⦇ sig ⦈ s'` is meant to be read as a partial pattern, so it introduces the variables `e`, `s`, `sig`, while `s'` is an expression. This isn't a thing in Agda, so what we do instead is to introduce those variables elsewhere and then use them here. We could have done other things, where the variables are properly introduced by a pattern, but then we don't get this nice presentation for the PDF.

A cleaner alternative (in terms of the Agda code) would be to put a `∀` on each constructor that introduces all those variables - then you don't have those large `variable` blocks that affect things for the entire file. However, then we need to put that `∀` on its own line with a `[hide]` block around it, which adds a few lines for each constructor.

One other approach might be to put everything in a `module _ where` block. Then the `variable` block only affects the STS constructors and nothing after it. Maybe that's the cleanest solution...

Reply:
I still find the `variable` approach by far the cleanest, assuming that consistent naming is already followed properly of course.